erpnext/controllers/stock_controller.py

```diff
@@ -331,15 +331,19 @@ def validate_inspection(self):
 
 		for d in self.get('items'):
 			raise_exception = False
-			if (inspection_required_fieldname and not d.quality_inspection and
-				frappe.db.get_value("Item", d.item_code, inspection_required_fieldname)):
-				raise_exception = True
-			elif self.doctype == "Stock Entry" and not d.quality_inspection and d.t_warehouse:
-				raise_exception = True
+			if not d.quality_inspection:
```

Comment:
This logic doesn't make sense.

If the QI doesn't exist then it should just raise. And if it doesn't exist why are we checking for other shit? What if it doesn't exist and the other ifs are not false? It'll go into the else and explode. Your logic is reverse 😅 

If the Item Requires Inspection, and if the QI is missing, then break. 
If the doctype is Stock Entry and Target Warehouse exists, and if the QI is missing, then break.

That's what we're trying to test for.